### PR TITLE
Combine CommentsPage logic into CommentsContainer

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game.Tests/Visual/Online/TestSceneOfflineCommentsContainer.cs [hunk 1]
```diff
@@ -3,90 +3,52 @@
...
-        }
+        });
 
         [Test]
-        public void TestEmptyBundle()
+        public void TestLocalCommentBundle()
         {
-            AddStep("Create page", () => createPage(getEmptyCommentBundle()));
-            AddAssert("Dictionary length is 0", () => commentsPage?.DictionaryLength == 0);
-        }
```
Comment (bdach) [inline osu.Game.Tests/Visual/Online/TestSceneOfflineCommentsContainer.cs]: What was the point of having these tests if they're going to get removed with this refactoring? The first one in particular was a *regression test*. This is unacceptable.

**Review by bdach — DISMISSED** [review]

File: osu.Game/Overlays/Comments/CommentsContainer.cs [hunk 2]
```diff
@@ -153,39 +156,44 @@ private void getComments()
...
-                Sort = { BindTarget = Sort },
-                Type = { BindTarget = type },
-                CommentableId = { BindTarget = id }
-            }, loaded =>
+                content.Add(new NoCommentsPlaceholder());
+                moreButton.Hide();
+                return;
+            }
+
+            var topLevelComments = AppendComments(response);
```
Comment (smoogipoo) [inline osu.Game/Overlays/Comments/CommentsContainer.cs]: Really not feeling this. Some comments are async-loaded while others parts aren't.

The loading part itself should be moved inside `AppendComments`. The other stuff done here can exist outside of the load.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Review by bdach — COMMENTED** [review]
otherwise looks ok I suppose

File: osu.Game.Tests/Visual/Online/TestSceneOfflineCommentsContainer.cs [hunk 3]
```diff
@@ -211,16 +178,18 @@ private void createPage(CommentBundle commentBundle)
...
-            {
-            }
-
             public new void AppendComments([NotNull] CommentBundle bundle) => base.AppendComments(bundle);
 
             public int DictionaryLength => CommentDictionary.Count;
+
+            public void ShowComments(CommentBundle bundle)
+            {
+                CommentCounter.Current.Value = 0;
```
[the remark this answers is not in the document]
Comment (EVAST9919) [reply osu.Game.Tests/Visual/Online/TestSceneOfflineCommentsContainer.cs:189]: `ClearComments` called even if we are switching the sort which looks a bit weird. (when it jumps from `number` to 0 and then to this exact number again)

**Comment by gagahpangeran** [comment]
What is the state of this PR?

I want to add pinned comment support and looks like this structure is easier to use.

**Review by smoogipoo — APPROVED** [review]
This can be refactored further, but I think it's fine to get in for the time being, so progress can be made in areas such as pinning.

**Pull request merged**